### ISSUE TITLE
Can't add superuser new install freebsd using PostgreSQL database

### ISSUE THREAD

**Issue by whitehome** [postfixadmin/postfixadmin#471]
I am installing Postfixadmin 3.3.8 which appears to be the latest version.  I tried installing 3.3.1 but had issues creating the superuser as well just different issues.  I am only going to deal with version 3.3.8 in this post.

I moved the code to my web directory and fixed up the permissions and added templates_c as directed.  I am using PostgreSQL 12.5 Freebsd 12 and Apache24.  I am also configuring postfix 3.5.8.1 and Dovecot 2.3.13 but I have postfix and dovecot stopped while I get the database configured and populated.  I was able to run postfixadmin setup which created the database tables and everything looked good until I got to creating the superuser admin account which fails.   My issues are as follows:

New from version 3.3.1 to 3.3.8 are the following warnings:
Warning: session_cache_limiter(): Cannot change cache limiter when headers already sent in /.../common.php on line 23
Warning: session_name(): Cannot change session name when headers already sent in /.../common.php on line 24
Warning: session_start(): Cannot start session when headers already sent in /.../html/common.php on line 25

These messages are at the beginning of the setup and setup progressed fine right up to including the add admin user ID which I performed but I got the following error after adding the admin ID:

Fatal error: Uncaught Error: Call to undefined function json_encode() in /.../public/setup.php:387 Stack trace: #0 {main} thrown in /.../public/setup.php on line 387

This error is from the error logic after the following statement returned a 1 (error):
list($error, $setupMessage, $errors) = create_admin($values);

I found the following entries in my messages log:

Mar 28 18:08:52 white-home postgres[79848]: [8-1] 2021-03-28 18:08:52.650 EDT [79848] ERROR:  relation "config" does not exist at character 19
Mar 28 18:08:52 white-home postgres[79848]: [8-2] 2021-03-28 18:08:52.650 EDT [79848] STATEMENT:  SELECT value FROM config WHERE name = 'version'
Mar 28 18:11:40 white-home postgres[79883]: [8-1] 2021-03-28 18:11:40.730 EDT [79883] ERROR:  language "plpgsql" already exists
Mar 28 18:11:40 white-home postgres[79883]: [8-2] 2021-03-28 18:11:40.730 EDT [79883] STATEMENT:  CREATE LANGUAGE plpgsql

Not sure if that aborted the create admin function or not.

I tried putting ?debug=1 on the setup.php statement but no other output was produced.

The INSTALL.txt didn't help other than for setup instructions.

If I knew the tables to update I would attempt to create the superuser by hand.  Maybe you can give me an older version number to use that will work and get me going.  I have used postfixadmin before but with mysql don't know if PostgreSQL is a weaker implementation or what is happening.  Any thoughts would be appreciated.

**Comment by DavidGoodwin** [comment]
Hi - I think this is fixed with https://github.com/postfixadmin/postfixadmin/commit/d5c14240a456c43fe9a53e0b01225f1cef43022b (it has just not made it's way into a release yet).

**Comment by DavidGoodwin** [comment]
(I think it only affects people using FPM or something like that which is stopping normal output buffering from soaking up the output)

**Comment by whitehome** [comment]
Thanks for the input, I already have that fix.

**Comment by DavidGoodwin** [comment]
The fatal error implies you're missing the 'json' module in your PHP environment. Are you able to install it?

**Comment by whitehome** [comment]
So that was it sort of, I had ruby-json but not php-json and was checking into that this morning already.  Installing php-json got me past setup, my real problem was I didn't have 2 numbers in my password but wasn't getting the error message.  When I tried to login with my new user I found I needed to install php-hash so other missing packages my follow but things are now working.  Never solved the cache issue in setup but that was only a Warning and not fatal.  Thanks for your help and patience.
